libexec/scoop-download.ps1

```diff
@@ -0,0 +1,127 @@
+if(!$use_cache) { warn "Cache is being ignored." }
+
+foreach ($curr_app in $apps) {
+    # Prevent leaking variables from previous iteration
+    $bucket = $version = $app = $manifest = $url = $null
+
+    $app, $bucket, $version = parse_app $curr_app
+    $app, $manifest, $bucket, $url = Find-Manifest $app $bucket
+
+    info "Starting download for $app"
+
+    # Generate manifest if there is different version in manifest
+    if (($null -ne $version) -and ($manifest.version -ne $version)) {
+        $generated = generate_user_manifest $app $bucket $version
+        if ($null -eq $generated) {
+            error 'Manifest cannot be generated with provided version'
+            continue
+        }
+        $manifest = parse_json($generated)
+    }
+
+    if(!$manifest) {
+        error "Couldn't find manifest for '$app'$(if($url) { " at the URL $url" })."
+        continue
+    }
+    $version = $manifest.version
+    if(!$version) { 
+        error "Manifest doesn't specify a version."
+        continue
+    }
+    if($version -match '[^\w\.\-\+_]') {
+        error "Manifest version has unsupported character '$($matches[0])'."
+        continue
+    }
+
+    $curr_check_hash = $check_hash
+    if ($version -eq 'nightly') {
+        $version = nightly_version $(get-date)
+        $curr_check_hash = $false
+    }
+
+    if(!(supports_architecture $manifest $architecture)) {
+        error "'$app' doesn't support $architecture architecture!"
+        continue
+    }
+
+    if(Test-Aria2Enabled) {
+        dl_with_cache_aria2 $app $version $manifest $architecture $cachedir $manifest.cookie $use_cache $curr_check_hash
+    } else {
+        foreach($url in script:url $manifest $architecture) {
+            try {
+                dl_with_cache $app $version $url $null $manifest.cookie $use_cache
+            } catch {
+                write-host -f darkred $_
+                error "URL $url is not valid"
+                continue
+            }
+
+            if($curr_check_hash) {
+                $manifest_hash = hash_for_url $manifest $url $architecture
+                $cached = cache_path $app $version $url
+                $ok, $err = check_hash $cached $manifest_hash (show_app $app $bucket)
+
+                if(!$ok) {
+                    error $err
+                    if(test-path $cached) {
+                        # rm cached file
+                        Remove-Item -force $cached
+                    }
+                    if ($url -like '*sourceforge.net*') {
+                        warn 'SourceForge.net is known for causing hash validation fails. Please try again before opening a ticket.'
+                    }
+                    error (new_issue_msg $app $bucket "hash check failed")
+                    continue
+                }
+            } else {
+                info "Skipping hash verification."
+            }
+        }
+    }
+
+    success "'$app' ($version) was downloaded successfully!"
```

Comment:
```suggestion
    success "'$app' ($version) was downloaded and hash checked successfully!"
```

Reply:
We have a flag for skipping hash check, so this will not be always true.

Reply:
Use an `if` to conditionally output different statement? Likes:

```pwsh
if ($curr_check_hash) {
    success "'$app' ($version) was downloaded and hash checked successfully!"
} else {
    success "'$app' ($version) was downloaded successfully!"
}
```

And IMO `--no-hash-check` is a better switch name. (instead of simple `--skip` which may indicate skip downloading.

Reply:
> Use an `if` to conditionally output different statement? Likes:

The information whether hash is being checked is already given by the `check_hashes` function, see: 
![image](https://user-images.githubusercontent.com/46838874/148338154-0b7bbda9-87fd-427e-bc7c-89a613f67125.png)

We should show it only once.

> And IMO `--no-hash-check` is a better switch name. (instead of simple `--skip` which may indicate skip downloading.

Will change it

Reply:
Also, the primary function of this command is to just `download` - hash check is a supplementary internal addition, useful for maintainers but not necessarily users.